website/src/components/Chat/ChatContext.tsx

```diff
@@ -0,0 +1,16 @@
+import { createContext, PropsWithChildren, useContext, useMemo } from "react";
+import { ModelInfo } from "src/types/Chat";
+
+export type ChatContext = {
+  modelInfos: ModelInfo[];
+};
+
+const chatContext = createContext<ChatContext>({} as ChatContext);
+
+export const useChatContext = () => useContext(chatContext);
+
+export const ChatContextProvider = ({ children, ...props }: PropsWithChildren<ChatContext>) => {
+  const value = useMemo(() => props, [props]);
```

Comment:
does this make any difference?

Reply:
I update to explicit memo the `modelInfos`. Haven't tested the performance, but it's recommended to memo context value.